business/checkers/gateways/multi_match_checker.go

```diff
@@ -86,8 +87,9 @@ func parsePortAndHostnames(serverDef map[string]interface{}) []Host {
 	if portDef, found := serverDef["port"]; found {
 		if ports, ok := portDef.(map[string]interface{}); ok {
 			if numberDef, found := ports["number"]; found {
-				if portNumber, ok := numberDef.(int64); ok {
-					port = int(portNumber)
+				portNumber, e := intutil.Convert(numberDef)
+				if e != nil {
```

Comment:
Isn't this incorrect in any case? If err == nil, then we can use the value. Otherwise it is the default value. And then merge the lines:

```go
if portNumber, err := intutil.Convert(numberDef); err == nil {
  port = portNumber
}
```

Reply:
You're right